packages/widget/src/context/wallet.ts

```diff
@@ -31,6 +33,10 @@ export enum WalletContextKeys {
   SUBSTRATE_WALLET = 'substrateWallet'
 }
 
+export interface SubstrateProviderContext {
+  substrateProvider?: ApiPromise;
```

Comment:
Q: shouldn't widget user be able to provide multiple providers for each parachain. Like what happens if user switches source chain?

Reply:
I need to test this

Reply:
possible solution here: change type to `Array<{ domainId: number, provider: ApiPromise }>`, user sets the amount of providers he wants, as long as they are supported by us -which does not depend on us-, upon source network selection we match agains the `domainId`, init the provider and build the asset. Wdyt?

Reply:
@mpetrunic I provided possible solution to this that also touches the comment below. Minor nitt here is that perhaps is better to use `chainId` instead of `domainId`